server/storage/backend/read_tx.go

```diff
@@ -120,6 +130,82 @@ func (baseReadTx *baseReadTx) UnsafeRange(bucketType Bucket, key, endKey []byte,
 	return append(k2, keys...), append(v2, vals...)
 }
 
+func bytesToRev(bytes []byte) int64 {
+	return int64(binary.BigEndian.Uint64(bytes[0:8]))
+}
+
+func (baseReadTx *baseReadTx) UnsafeRangeWithLock(bucketType Bucket, key, endKey []byte, limit int64, b Backend) ([][]byte, [][]byte) {
```

Comment:
Can you add documentation to this function? I'm also not sure about it's name "Unsafe" prefix is used as a convention that it's caller responsibility to lock the transaction, not sure what `UnsafeRageWithLock` means.

Reply:
UnsafeRange do not lock the transaction either. 
I think "Unsafe" prefix means  Rlock of concurrentReadTx, and Runlock at the end of transaction:

```
if tr.readMode == ConcurrentReadTxNoCopyMode && tr.tx == nil { // ConcurrentReadTxNoCopyMode
		...

		tx.RLock() // RLock is no-op. concurrentReadTx does not need to be locked after it is created.
		tr.tx = tx
	}

func (tr *storeTxnRead) End() {
	if tr.tx != nil {
		tr.tx.RUnlock() // RUnlock signals the end of concurrentReadTx.
	}
	tr.s.mu.RUnlock()
}
```